autometa/common/coverage.py

```diff
@@ -60,16 +60,33 @@ def from_spades_names(records):
         Why the exception is raised.
 
     """
-    return pd.Series(
+    logger.info(f'Retrieving coverages from contig ID in {args.assembly}')
+    coverages = pd.Series(
         {record.id:record.id.split('_cov_')[-1] for record in records},
         name='coverage',
         dtype=float)
+    coverages.index.name = 'contig'
+    return coverages
 
 def make_length_table(fasta, out):
-    seqs = {r.id:len(r) for r in SeqIO.parse(fasta, 'fasta')}
-    length_s = pd.Series(seqs, name='length')
-    length_s.index.name = 'contig'
-    length_s.to_csv(out, sep='\t', index=True, header=True)
+    """Writes a tab-delimited length table to `out` given an input `fasta`.
+
+    Parameters
+    ----------
+    fasta : str
+        </path/to/assembly.fasta>
+    out : str
+        </path/to/lengths.tsv>
+
+    Returns
+    -------
+    str
+        </path/to/lengths.tsv>
+    """
+    seqs = {record.id:len(record) for record in SeqIO.parse(fasta, 'fasta')}
+    lengths = pd.Series(seqs, name='length')
+    lengths.index.name = 'contig'
+    lengths.to_csv(out, sep='\t', index=True, header=True)
     return out
 
 def get(fasta, out, fwd_reads=None, rev_reads=None, sam=None, bam=None, lengths=None,
```

Comment:
If I'm not mistaken this function still does not handle single end data (issue #21, item 4).

Reply:
A `kwargs` keyword argument is allowed within the bowtie script.

See:

1. [bowtie2 align function definition](https://github.com/WiscEvan/Autometa/blob/5c735796764df761558d5035bfd7248409a5fe1a/autometa/common/external/bowtie.py#L83-L101 "bowtie align function definition with docstring")
2. [adding additional options](https://github.com/WiscEvan/Autometa/blob/5c735796764df761558d5035bfd7248409a5fe1a/autometa/common/external/bowtie.py#L115)

single-end (as well as other) options can be passed in via `kwargs`

Reply:
But it throws an error if it has `fwd_reads` but not `rev_reads`...

Reply:
From the `bowtie.align` definition it looks like it always includes `-1` and `-2`, so it will now work if either fwd or rev are not defined.

Reply:
I believe I've addressed the bowtie.py issues you raised? See [here](https://github.com/KwanLab/Autometa/blob/406041bcbca9e9e14fc3991cad8ad5104509de83/autometa/common/external/bowtie.py#L83-L137)